authz/providers/azure/rbac/rbac.go

```diff
@@ -324,5 +383,8 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 	}
 
 	// Decode response and prepare k8s response
-	return ConvertCheckAccessResponse(data)
+	reviewResult.status, reviewResult.err = ConvertCheckAccessResponse(data)
+	ch <- reviewResult
+
+	wg.Done()
```

Comment:
i think you should use defer to invoke wg.Done()

Reply:
added at the start of the func